modules/phase_field/src/action/GrandPotentialKernelAction.C

```diff
@@ -244,50 +262,136 @@ GrandPotentialKernelAction::act()
+        params.set<NonlinearVariableName>("variable") = w_names[i];
+        params.set<std::vector<VariableName>>("v") = notarealvector;
+        params.set<std::vector<VariableName>>("args") = v0;
+        params.set<std::vector<MaterialPropertyName>>("Fj_names") = fj_temp;
+        params.set<std::vector<MaterialPropertyName>>("hj_names") = hj;
+        params.set<bool>("implicit") = implicity;
+        params.set<bool>("use_displaced_mesh") = displaced_mesh;
+        kernel_name = "Coupled_" + w_names[i] + "_" + all_etas[j];
+        _problem->addKernel("CoupledSwitchingTimeDerivative", kernel_name, params);
+      } //     for (unsigned int j = 0; j < n_etas + n_grs; ++j) coupledSwitchingTimeDeriv
+    } //for (unsigned int i = 0; i < n_w; ++i) chempot
+    } // case false
+    case true: // mass conservation kernels with conc and chempot coupling
+    {
+      std::vector<VariableName> v3;
+      v3.resize(n_w);
+      for (unsigned int i = 0; i < n_w; ++i)
+        v3[i] = w_names[i];
+      unsigned int n_c = c_names.size();
+      std::vector<VariableName> v4;
+      v4.resize(n_c);
+      for (unsigned int i=0; i < n_c; ++i)
+        v4[i] = c_names[i];
+      for (unsigned int i = 0; i < n_c; ++i)
+      {
+        // TimeDerivative concentration Kernel
+        InputParameters params = _factory.getValidParams("TimeDerivative");
+        params.set<NonlinearVariableName>("variable") = c_names[i];
+        params.set<bool>("implicit") = implicity;
+        params.set<bool>("use_displaced_mesh") = displaced_mesh;
+        kernel_name = "DT_" + c_names[i];
+        _problem->addKernel("TimeDerivative", kernel_name, params);
+     
+        // MatDiffusion concentration (coupled with chempot)
+        params = _factory.getValidParams("MatDiffusion");
+        params.set<NonlinearVariableName>("variable") = c_names[i];
+        params.set<std::vector<VariableName>>("v") = v3;
+        params.set<bool>("implicit") = implicity;
+        params.set<bool>("use_displaced_mesh") = displaced_mesh;
+        params.set<MaterialPropertyName>("diffusivity") = M[i];
+        kernel_name = "MatDif_" + w_names[i];
+        if (aniso.get(i))
+        {
+          params.set<std::vector<VariableName>>("args") = v1;
+          _problem->addKernel("MatAnisoDiffusion", kernel_name, params);
+        }
+        else
+        {
+           params.set<std::vector<VariableName>>("args") = v0;
+           _problem->addKernel("MatDiffusion", kernel_name, params);
+        }
+      } // for (unsigned int i = 0; i < n_c; ++i)
+      
+      // Chemical Potentials
+      for (unsigned int i = 0; i < n_w; ++i)
+      {
+        // coupling of c and w
+        InputParameters params = _factory.getValidParams("MatReaction");
+        params.set<NonlinearVariableName>("variable") = w_names[i];
+        params.set<std::vector<VariableName>>("v") = v4;
+        params.set<MaterialPropertyName>("mob_name") = mob_cw;
```

Comment:
Here we just want `mob_name` to be a material property that is `-1` everywhere. Right now it is defined and calculated in `GrandPotentialSinteringMaterial`, but we create a lot of extra variables doing that. I can't find an example that would show that we could just define it as `-1` here, maybe @dschwen might have a suggestion?

Reply:
Just set it to `"-1"`